nemoguardrails/embeddings/basic.py

```diff
@@ -267,9 +270,32 @@ async def search(self, text: str, max_results: int = 20) -> List[IndexItem]:
         else:
             _embedding = (await self._get_embeddings([text]))[0]
 
+        if self._index is None:
+            raise ValueError(
+                "Index is not built yet. Ensure to call `build` before searching."
+            )
+
         results = self._index.get_nns_by_vector(
             _embedding,
             max_results,
+            include_distances=True,
         )
 
-        return [self._items[i] for i in results]
+        filtered_results = self._filter_results(
+            results[0], results[1], self.search_threshold
+        )
+
+        return [self._items[i] for i in filtered_results]
+
+    @staticmethod
+    def _filter_results(
+        indices: List[int], distances: List[float], threshold: float
+    ) -> List[int]:
+        if threshold == float("inf"):
+            return indices
+        else:
+            return [
+                index
+                for index, distance in zip(indices, distances)
+                if distance <= threshold
```

Comment:
The `threshold` should be a similarity threshold, with values between 0 and 1. We can convert from the distance provided by Annoy to a similarity threshold using the formula `1 - distance / 2` (the distance is between 0 and 2).